ner_v2/detectors/temporal/date/date_detection.py

```diff
@@ -322,21 +327,24 @@ def _detect_return_date(self):
         """
 
         date_dict_list = []
-        regex_pattern_1 = re.compile(r'\s?((coming back|back|return date\:?|return date -|returning on|'
-                                     r'arriving|arrive|return|returning at)\s+(.+))\.?\s?', flags=re.UNICODE)
-        regex_pattern_2 = re.compile(r'(.+)\s+(ko|k|)\s*(aana|ana|aunga|aaun)', flags=re.UNICODE)
-        patterns_1 = regex_pattern_1.findall(self.processed_text.lower())
-        patterns_2 = regex_pattern_2.findall(self.processed_text.lower())
-        if patterns_1:
-            for pattern in patterns_1:
-                date_dict_list.extend(
-                    self._date_dict_from_text(text=pattern[2], to_property=True)
-                )
-        elif patterns_2:
-            for pattern in patterns_2:
-                date_dict_list.extend(
-                    self._date_dict_from_text(text=pattern[0], to_property=True)
-                )
+        regex_pattern_1 = re.compile(r'\b'
+                                     r'(?:check(?:\s|\-)?out date (?:is|\:)?|'
+                                     r'coming back|return date\s?(?:\:|\-)?|returning on|returning at|'
+                                     r'arriving|arrive|return|back)'
+                                     r'\s+(.+?)(?:\band|&|(?<!\d)\.|$)', flags=re.UNICODE)
+        regex_pattern_2 = re.compile(r'(.+?)\s+(?:ko?\s+)?(?:aana|ana|aunga|aaun)', flags=re.UNICODE)
+        matches = None
+        matches_1 = regex_pattern_1.findall(self.processed_text)
+        matches_2 = regex_pattern_2.findall(self.processed_text)
+
+        if matches_1:
+            matches = matches_1
+        elif matches_2:
+            matches = matches_2
+
+        matches = matches or []
+        for match in matches:
+            date_dict_list.extend(self._date_dict_from_text(text=match, to_property=True))
```

Comment:
Did we test this? 
Input to 'self._date_dict_from_text' is changing.

Reply:
Yes this works as now only one groups is selected, match is no longer a tuple